src/Core/Framework/DataAbstractionLayer/Write/EntityWriteResultFactory.php

```diff
@@ -358,9 +358,8 @@ private function buildQueueResults(WriteCommandQueue $queue): array
             $jsonUpdateCommands = [];
             $writeResults = [];
 
-            foreach ($commands as $command) {
+            foreach ($commands as $uniqueId => $command) {
                 $primaryKey = $this->getCommandPrimaryKey($command, $primaryKeys);
-                $uniqueId = \is_array($primaryKey) ? implode('-', $primaryKey) : $primaryKey;
 
                 if ($command instanceof JsonUpdateCommand) {
```

Comment:
Hey, thanks for the contribution.

I see what you are trying do fix here, but this breaks the "merge json+normal" logic where the unique ID is used for.

I would appriciate, when you just merge the payloads here:


```
$results = [];

foreach ($commands as $command) {
   $payload = $command->getPayload();
   
   $uniqueId = $command->getPrimaryKey();

   .... 


   $results[$uniqueId] = new EntityWriteResult(
      payload: array_merge($payload, $results[$uniqueId] ?? []) 
      ...
   );
}


foreach ($json as $command) {
   $payload = $command->getPayload();
   
   $uniqueId = $command->getPrimaryKey();

   .... 

   $results[$uniqueId] = new EntityWriteResult(
      payload: array_merge($payload, $results[$uniqueId] ?? []) 
      ...
   );
}
```

In this case you get one single write result for each entity.